bundle/edu.gemini.phase2.skeleton.servlet/src/test/scala/edu/gemini/phase2/skeleton/factory/TemplateSpec.scala

```diff
@@ -0,0 +1,168 @@
+package edu.gemini.phase2.skeleton.factory
+
+import java.security.Principal
+
+import edu.gemini.model.p1.immutable.{Target, NifsBlueprintBase, BlueprintBase, TimeAmount, Condition, Observation, Magnitude, SiderealTarget, Proposal}
+import edu.gemini.model.p1.mutable.{MagnitudeSystem, MagnitudeBand }
+import edu.gemini.phase2.core.model.SkeletonShell
+import edu.gemini.phase2.core.odb.SkeletonStoreService
+import edu.gemini.phase2.template.factory.api.TemplateFolderExpansionFactory
+import edu.gemini.phase2.template.factory.impl.{TemplateDb, TemplateFactoryImpl}
+import edu.gemini.pot.sp.{ISPTemplateGroup, ISPProgram}
+import edu.gemini.pot.spdb.DBLocalDatabase
+import edu.gemini.shared.skyobject.Magnitude.Band
+import edu.gemini.spModel.core.SPProgramID
+import edu.gemini.spModel.obscomp.SPNote
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.obs.SPObservation
+import edu.gemini.spModel.template.{TemplateParameters }
+import org.specs2.mutable.Specification
+import scala.collection.JavaConverters._
+import scalaz._, Scalaz._
+import Observation.{blueprint, target, condition, time}
+import Proposal.{ targets, observations }
+
+/** 
+ * Mixin for a specs test for template expansion. This provides code to expand a Phase 1 Proposal
+ * into a skeleton, then test that the expansion was correct.
+ * @param xmlName name of the template to test: "NIFS_BP.xml" for example
+ */
+abstract class TemplateSpec(xmlName: String) { this: Specification =>
+
+  // Required when constructing phase 1 proposals
+  if (System.getProperty("edu.gemini.model.p1.schemaVersion") == null)
```

Comment:
There may be a better way to do this, but I was unable to construct a `Proposal` without it.